qiskit_dynamics/models/operator_collections.py

```diff
@@ -124,138 +153,299 @@ class SparseOperatorCollection(BaseOperatorCollection):
+    def static_dissipators(self) -> Array:
+        """Returns operators for static part of dissipator."""
+
+    @static_dissipators.setter
+    @abstractmethod
+    def static_dissipators(self, new_static_dissipators: Optional[Array] = None):
+        """Sets operators for static part of dissipator."""
+
+    @property
+    @abstractmethod
+    def dissipator_operators(self) -> Array:
+        """Returns operators for non-static part of dissipator."""
+
+    @dissipator_operators.setter
+    @abstractmethod
+    def dissipator_operators(self, new_dissipator_operators: Optional[Array] = None):
+        """Sets operators for non-static part of dissipator."""
+
+    @abstractmethod
+    def evaluate_hamiltonian(self, ham_sig_vals: Union[None, Array]) -> Union[csr_matrix, Array]:
+        """Evaluate the Hamiltonian of the model."""
+
+    @abstractmethod
+    def evaluate(
+        self, ham_sig_vals: Union[None, Array], dis_sig_vals: Union[None, Array]
+    ) -> Union[csr_matrix, Array]:
+        r"""Evaluate the map."""
+
+    @abstractmethod
+    def evaluate_rhs(
+        self, ham_sig_vals: Union[None, Array], dis_sig_vals: Union[None, Array], y: Array
+    ) -> Array:
+        r"""Compute the function."""
 
-        self._hamiltonian_operators = to_array(hamiltonian_operators)
-        self._dissipator_operators = to_array(dissipator_operators)
-        if dissipator_operators is not None:
-            self._dissipator_operators_conj = np.conjugate(
-                np.transpose(to_array(dissipator_operators), [0, 2, 1])
+    def __call__(
+        self, ham_sig_vals: Union[None, Array], dis_sig_vals: Union[None, Array], y: Optional[Array]
+    ) -> Union[csr_matrix, Array]:
+        """Evaluate the model, or evaluate the RHS."""
+        if y is None:
+            return self.evaluate(ham_sig_vals, dis_sig_vals)
+
+        return self.evaluate_rhs(ham_sig_vals, dis_sig_vals, y)
+
+    def copy(self):
+        """Return a copy of self."""
+        return copy(self)
+
+
+class DenseLindbladCollection(BaseLindbladOperatorCollection):
+    r"""Object for computing the right hand side of the Lindblad equation
+    with dense arrays.
+    """
+
+    @property
+    def static_hamiltonian(self) -> Array:
+        return self._static_hamiltonian
+
+    @static_hamiltonian.setter
+    def static_hamiltonian(self, new_static_hamiltonian: Optional[Array] = None):
+        self._static_hamiltonian = to_array(new_static_hamiltonian)
+
+    @property
+    def hamiltonian_operators(self) -> Array:
+        return self._hamiltonian_operators
+
+    @hamiltonian_operators.setter
+    def hamiltonian_operators(self, new_hamiltonian_operators: Optional[Array] = None):
+        self._hamiltonian_operators = to_array(new_hamiltonian_operators)
+
+    @property
+    def static_dissipators(self) -> Array:
+        return self._static_dissipators
+
+    @static_dissipators.setter
+    def static_dissipators(self, new_static_dissipators: Optional[Array] = None):
```

Comment:
The model classes would have same signature as these collection classes if you can make `in_frame_basis` a property rather than kwarg